keras_hub/src/layers/modeling/rotary_embedding.py

```diff
@@ -113,51 +180,231 @@ def _compute_positions(self, inputs, start_index=0):
         return positions + ops.cast(start_index, dtype="float32")
 
     def _compute_cos_sin_embedding(self, inputs, start_index=0, positions=None):
+        """Compute cos & sin RoPE embeddings with optional YaRN scaling.
+        Uses tensor ops only to remain JIT/backends friendly.
+        """
         batch_axis = 0
-        feature_axis = len(inputs.shape) - 1
         sequence_axis = 1
+        feature_axis = len(inputs.shape) - 1
 
+        # rotary_dim should be half of the last
+        # feature axis (HF-style: rotate pairs)
         rotary_dim = ops.shape(inputs)[feature_axis]
+        # Validate evenness
+        try:
+            # best-effort check when running eagerly;
+            # if unavailable this will be a no-op
+            if int(rotary_dim) % 2 != 0:
+                raise ValueError(
+                    "Rotary embedding requires even feature "
+                    "dimension (last axis)."
+                )
+        except Exception:
+            pass
+
+        # Get inverse frequencies using the appropriate
+        # scaling method (linear, dynamic, yarn, etc.)
```

Comment:
Let's not use try except block, can't we just use _validate_rotary_dimension function? 
Also, remove the comments.